pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1626,7 +1628,8 @@ func (r *ReconcileClusterDeployment) ensureManagedDNSZone(cd *hivev1.ClusterDepl
 		message = dnsErrorCondition.Message
 	default:
 		status = corev1.ConditionTrue
-		reason, message, requeueAfter = dnsZoneNotReadyMaybeTimedOut(cd, cdLog)
+		// If dnsZoneNotReadyMaybeTimedOut, then ProvisionStopped should have been set to true
+		reason, message, requeueAfter, setProvisionFailedTerminalMetric = dnsZoneNotReadyMaybeTimedOut(cd, cdLog)
```

Comment:
...so the bool return here would be fed into L1664 just to decide whether to do a status update. But it would not be used to decide whether to increment the metric.